### PR TITLE
Issue 201

### PR DESCRIPTION
**Pull request by gregw**
Some logging improvements and a NPE fix found in working on #201

### CODE REVIEW

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 1]
```diff
@@ -196,7 +196,8 @@ public void save() {
 
     protected void save (boolean force)
     {
-      logger.fine("Session " + getId() + "is"+(dirty?" dirty":" not dirty")+(force || dirty ? " saving":" not saving"));
+      if (logger.isLoggable(Level.FINE))
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: http://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 2]
```diff
@@ -196,7 +196,8 @@ public void save() {
...
+        logger.fine("Session " + getId() + "is"+(dirty?" dirty":" not dirty")+(force || dirty ? " saving":" not saving"));
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: I would consider using formatter for such a string.

I assume `getId()` does not have to end with space and therefore you probably want to change "is" to " is"

spaces around " + ", " ? " and " : " - http://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 3]
```diff
@@ -216,6 +217,8 @@ protected void save (boolean force)
                     sessionStore.saveSession(key, sessionData);
                   }
                   dirty = false;
+                  if (logger.isLoggable(Level.FINE))
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: braces and should it be " saved"?

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 4]
```diff
@@ -316,8 +319,9 @@ protected boolean access(long accessTime) {
       if (dirty) {
       } else if (timeRemaining < (getSessionExpirationInMilliseconds() * UPDATE_TIMESTAMP_RATIO)) {
         dirty = true;
-        logger.fine("Session " + getId() + " accessed while near expiration, marking dirty.");
-      } else {
+        if (logger.isLoggable(Level.FINE))
+          logger.fine("Session " + getId() + " accessed while near expiration, marking dirty.");
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: braces

Also, any reason for not being consistent with periods?

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 5]
```diff
@@ -411,12 +415,16 @@ SessionData loadSession(String sessionId) {
     String key = SESSION_PREFIX + sessionId;
 
     SessionData data = null;
+    if (logger.isLoggable(Level.FINE))
+      logger.fine("loadSession " + sessionId);
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: braces

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 6]
```diff
@@ -411,12 +415,16 @@ SessionData loadSession(String sessionId) {
...
     for (SessionStore sessionStore : sessionStoresInReadOrder) {
       // Keep iterating until we find a store that has the session data we
       // want.
       try {
         data = sessionStore.getSession(key);
         if (data != null) {
+          if (logger.isLoggable(Level.FINE))
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: braces and space around " + "

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 7]
```diff
@@ -430,16 +438,17 @@ SessionData loadSession(String sessionId) {
     }
     if (data != null) {
       if (System.currentTimeMillis() > data.getExpirationTime()) {
-        logger.fine("Session " + sessionId + " expired " +
-                    ((System.currentTimeMillis() - data.getExpirationTime()) / 1000) +
-                    " seconds ago, ignoring.");
+        if (logger.isLoggable(Level.FINE))
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: braces

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 8]
```diff
@@ -430,16 +438,17 @@ SessionData loadSession(String sessionId) {
...
+          logger.fine("Session " + sessionId + " expired " + ((System.currentTimeMillis() - data.getExpirationTime()) / 1000) + " seconds ago, ignoring.");
         return null;
       }
     }
     return data;
   }
 
   SessionData createSession(String sessionId) {
+    if (logger.isLoggable(Level.FINE))
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: braces

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 9]
```diff
@@ -193,7 +195,7 @@ protected void startWebapp() throws Exception
       super.startWebapp();
     else
     {
-      LOG.info("Generating quickstart web.xml: {}", quickstartWebXml);
+      logger.info("Generating quickstart web.xml: "+quickstartWebXml);
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: " + "

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 10]
```diff
@@ -341,6 +343,8 @@ public boolean isTrustedRemoteAddr(String remoteAddr) {
   }
 
   public RequestContext getRequestContext(Request baseRequest) {
+    if (baseRequest==null)
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: braces and space around " == "

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 11]
```diff
@@ -381,26 +385,40 @@ public Interceptor getNextInterceptor() {
     VmApiProxyEnvironment getRequestSpecificEnvironment() {
       return requestSpecificEnvironment;
     }
+    
+    @Override
+    public String toString()
+    {
+      return String.format("RequestContext@%x %s==%s",hashCode(),request.getRequestURI(),requestSpecificEnvironment);
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: spaces after a comma

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 12]
```diff
@@ -381,26 +385,40 @@ public Interceptor getNextInterceptor() {
...
-      VmApiProxyEnvironment environment=requestContext.getRequestSpecificEnvironment();
-      String traceId=environment.getTraceId();
-      if (traceId!=null)
-        LogContext.current().put("traceId", environment.getTraceId());
-      if (LOG.isDebugEnabled())
-        LOG.debug("Enter {} -> {}",ApiProxy.getCurrentEnvironment(),environment);
-      ApiProxy.setEnvironmentForCurrentThread(environment); 
+      if (requestContext==null) {
+        if (logger.isLoggable(Level.FINE))
+          logger.fine("Entered with default env");
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: braces. 

Also, is it worth doing the check when you are using one constant string?

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 13]
```diff
@@ -381,26 +385,40 @@ public Interceptor getNextInterceptor() {
...
     public void enterScope(org.eclipse.jetty.server.handler.ContextHandler.Context context, Request baseRequest, Object reason) {
       RequestContext requestContext = getRequestContext(baseRequest);
-      VmApiProxyEnvironment environment=requestContext.getRequestSpecificEnvironment();
-      String traceId=environment.getTraceId();
-      if (traceId!=null)
-        LogContext.current().put("traceId", environment.getTraceId());
-      if (LOG.isDebugEnabled())
-        LOG.debug("Enter {} -> {}",ApiProxy.getCurrentEnvironment(),environment);
-      ApiProxy.setEnvironmentForCurrentThread(environment); 
+      if (requestContext==null) {
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: " == "

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 14]
```diff
@@ -381,26 +385,40 @@ public Interceptor getNextInterceptor() {
...
+        if (logger.isLoggable(Level.FINE))
+          logger.fine("Entered with default env");
+      } else {
+        VmApiProxyEnvironment environment=requestContext.getRequestSpecificEnvironment();
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: " = "

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 15]
```diff
@@ -381,26 +385,40 @@ public Interceptor getNextInterceptor() {
...
+      }
     }
 
     @Override
     public void requestInitialized(ServletRequestEvent sre) {
       ServletRequest request = sre.getServletRequest();
       Request baseRequest = Request.getBaseRequest(request);
+      
       RequestContext requestContext = getRequestContext(baseRequest);
+      if (logger.isLoggable(Level.FINE))
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: braces + " + "

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 16]
```diff
@@ -413,6 +431,8 @@ public void requestInitialized(ServletRequestEvent sre) {
     public void requestDestroyed(ServletRequestEvent sre) {      
       ServletRequest request = sre.getServletRequest();
       Request baseRequest = Request.getBaseRequest(request);
+      if (logger.isLoggable(Level.FINE))
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: like above

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 17]
```diff
@@ -429,18 +449,24 @@ public void onComplete(AsyncEvent event) throws IOException {
...
       }
     }
-    
+
     @Override
     public void exitScope(org.eclipse.jetty.server.handler.ContextHandler.Context context, Request baseRequest) {
-      if (LOG.isDebugEnabled())
-        LOG.debug("Exit {} -> {}",ApiProxy.getCurrentEnvironment(),defaultEnvironment);
+      if (logger.isLoggable(Level.FINE)) {
+        if (baseRequest==null)
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: braces, spaces around equality sign and the plus sign.

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 18]
```diff
@@ -429,18 +449,24 @@ public void onComplete(AsyncEvent event) throws IOException {
...
+      }
       ApiProxy.setEnvironmentForCurrentThread(defaultEnvironment);
       LogContext.current().remove("traceId");
     }
-    
+
     private void complete(Request baseRequest)
     {
       RequestContext requestContext = getRequestContext(baseRequest);
+      if (logger.isLoggable(Level.FINE))
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: like above.

File: testwebapp/src/main/java/com/google/apphosting/tests/usercode/testservlets/DumpServlet.java [hunk 19]
```diff
@@ -0,0 +1,116 @@
+package com.google.apphosting.tests.usercode.testservlets;
```
Comment (aozarov) [inline testwebapp/src/main/java/com/google/apphosting/tests/usercode/testservlets/DumpServlet.java]: copyright.

**Comment by gregw** [comment]
@aozarov Is there an eclipse formatting configuration file that has the full google style available?

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java (hunk 2 above)
Comment (gregw) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: I inherited the "" + "" style formatting.  Will change to more consistent String.format

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java (hunk 4 above)
Comment (gregw) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: inherited code, so have not looked in detail at periods logic or consistency

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java (hunk 5 above)
Comment (gregw) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: adding everywhere

**Comment by gregw** [comment]
I signed it!

**Comment by aozarov** [comment]
> @aozarov Is there an eclipse formatting configuration file that has the full google style available?

Yes, see [here](https://github.com/google/styleguide/blob/gh-pages/eclipse-java-google-style.xml).

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 20]
```diff
@@ -187,16 +187,15 @@ public void renewId(HttpServletRequest request) {
...
     public void save() {
       save(false);
     }
-    
 
     protected void save (boolean force)
     {
-      logger.fine("Session " + getId() + "is"+(dirty?" dirty":" not dirty")+(force || dirty ? " saving":" not saving"));
+      if (logger.isLoggable(Level.FINE)) {
+        logger.fine(String.format("Session %s dirty=%b force=%b", getId(),dirty,force));
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: I see some more formatting issues, like the spaces after a comma.
Will hold any formatting related comments until you run the formatter on the code.

File: testwebapp/src/main/java/com/google/apphosting/tests/usercode/testservlets/DumpServlet.java [hunk 21]
```diff
@@ -0,0 +1,132 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+@SuppressWarnings("serial")
+public class DumpServlet extends HttpServlet
+{
+    @Override
+    protected void doGet( HttpServletRequest request,
```
Comment (aozarov) [inline testwebapp/src/main/java/com/google/apphosting/tests/usercode/testservlets/DumpServlet.java]: I hope the formatter would fix the wrapping. It expect it to fix the spaces after "(" and before ")"  but if it does not fix the wrapping the prefered style would be:

```java
protected void doGet(HttpServletRequest request, HttpServletResponse response)
    throws ServletException, IOException {
````

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 22]
```diff
@@ -229,15 +231,12 @@ protected void save (boolean force)
...
             delay *= 2;
           }
-          logger.log(Level.SEVERE, "Unable to save session " + getId() +
-              " - too many attempts");
+          logger.log(Level.SEVERE, String.format("Session %s not saved: too many attempts", getId()));
         } catch (DeadlineExceededException e) {
-          logger.log(Level.SEVERE, "Unable to save session " + getId() +
-              " - too many timeouts.", e);
-        }
+          logger.log(Level.SEVERE, String.format("Session %s not saved: too many timeouts", getId()), e);        }
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: wrap closing '}'

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 23]
```diff
@@ -411,28 +412,33 @@ SessionData loadSession(String sessionId) {
...
         break;
       }
     }
     if (data != null) {
       if (System.currentTimeMillis() > data.getExpirationTime()) {
-        logger.fine("Session " + sessionId + " expired " +
-                    ((System.currentTimeMillis() - data.getExpirationTime()) / 1000) +
-                    " seconds ago, ignoring.");
+        if (logger.isLoggable(Level.FINE)) {
+          logger.fine(String.format("Session %s expired %d seconds ago, ignoring", sessionId,(System.currentTimeMillis() - data.getExpirationTime()) / 1000));
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: This line probably exceeds 100 characters.

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 24]
```diff
@@ -528,8 +533,7 @@ public void renewSessionId(String oldClusterId, String oldNodeId, String newClus
    */
   public void callSessionIdListeners (AbstractSession session, String oldId) {
     HttpSessionEvent event = new HttpSessionEvent(session);
-    for (HttpSessionIdListener l:_sessionIdListeners)
-    {
+    for (HttpSessionIdListener l:_sessionIdListeners) {
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: space before and after the colon. Also `listener` would be nicer than `l`

**Comment by gregw** [comment]
I ran the automated formatter on the touched files, so there are few more changes in areas that have not had behaviour changed.  But then this is not a very substantive PR, so it is mostly just cleanup, improved logging and one NPE fix.   Perhaps the NPE should be in it's own PR?

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 25]
```diff
@@ -189,14 +191,14 @@ protected void doStart() throws Exception {
   @Override
   protected void startWebapp() throws Exception
   {
-    if (quickstartWebXml==null)
+    if (quickstartWebXml==null) {
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: spaces around equality sign.

**Comment by aozarov** [comment]
> Perhaps the NPE should be in it's own PR?

Yes, Good idea.

**Comment by gregw** [comment]
I've created #218 with just the NPE fix.  Once that is merged, I'll update this to be just a logging/formatting cleanup.

File: testwebapp/src/main/java/com/google/apphosting/tests/usercode/testservlets/DumpServlet.java [hunk 26]
```diff
@@ -0,0 +1,133 @@
...
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+@SuppressWarnings("serial")
+public class DumpServlet extends HttpServlet {
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+    Collection<String> history = (Collection<String>) request.getAttribute("history");
+    if (history != null)
```
Comment (aozarov) [inline testwebapp/src/main/java/com/google/apphosting/tests/usercode/testservlets/DumpServlet.java]: The formatter didn't complain about the missing brackets?
Regardless, they should be fixed to be comply with the style guide.

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 27]
```diff
@@ -341,24 +333,26 @@ public boolean isTrustedRemoteAddr(String remoteAddr) {
   }
 
   public RequestContext getRequestContext(Request baseRequest) {
-    RequestContext requestContext = (RequestContext)baseRequest.getAttribute(RequestContext.class.getName());
-    if (requestContext==null) {
+    if (baseRequest == null)
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: brackets

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 28]
```diff
@@ -311,13 +303,13 @@ public void init(String appengineWebXmlFile)
...
-      AppEngineWebXmlReader appEngineWebXmlReader
-              = new AppEngineWebXmlReader(appDir, appengineWebXmlFile);
+      AppEngineWebXmlReader appEngineWebXmlReader =
+          new AppEngineWebXmlReader(appDir, appengineWebXmlFile);
       appEngineWebXml = appEngineWebXmlReader.readAppEngineWebXml();
     }
     VmRuntimeUtils.installSystemProperties(defaultEnvironment, appEngineWebXml);
     String logConfig = System.getProperty("java.util.logging.config.file");
-    if (logConfig!=null && logConfig.startsWith("WEB-INF/"))
+    if (logConfig != null && logConfig.startsWith("WEB-INF/"))
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: Braces

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 29]
```diff
@@ -160,62 +156,59 @@ protected void doStart() throws Exception {
...
   }
 
 
   @Override
-  protected void stopWebapp() throws Exception
-  {
-      if (quickstartWebXml==null)
-        super.stopWebapp();
+  protected void stopWebapp() throws Exception {
+    if (quickstartWebXml == null)
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: Braces

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 30]
```diff
@@ -160,62 +156,59 @@ protected void doStart() throws Exception {
...
     if (qswebxml.exists()) {
       setConfigurationClasses(quickstartConfigurationClasses);
     }
-    
+
     datastoreService = getDatastoreService();
-    
-    if (quickstartWebXml==null)
+
+    if (quickstartWebXml == null)
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: Braces (for the `else` as well).

File: jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java [hunk 31]
```diff
@@ -160,62 +156,59 @@ protected void doStart() throws Exception {
...
       addEventListener(new ContextListener());
     else
-      getMetaData().addDescriptorProcessor(preconfigProcessor=new PreconfigureDescriptorProcessor());
-    
+      getMetaData()
+          .addDescriptorProcessor(preconfigProcessor = new PreconfigureDescriptorProcessor());
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java]: putting the assignment in its own line would be preferred.

File: jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java [hunk 32]
```diff
@@ -316,12 +308,15 @@ protected boolean access(long accessTime) {
...
+        if (logger.isLoggable(Level.FINE)) {
+          logger.fine(
+              String.format("Session %s accessed while near expiration, marking dirty.", getId()));
+        }
+      } else if (logger.isLoggable(Level.FINE)) {
+        logger.fine(String.format("Session %s accessed early, not marking dirty.", getId()));
       }
-      sessionData.setExpirationTime(System.currentTimeMillis()
-              + getSessionExpirationInMilliseconds());
+      sessionData
```
Comment (aozarov) [inline jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java]: I am surprised to this that the formatter did that... I don't think that is desired but fine with as is.

**Comment by aozarov** [comment]
Looks good. Please fix merge conflicts and few more missing braces and we can merge it.

File: testwebapp/src/main/java/com/google/apphosting/tests/usercode/testservlets/DumpServlet.java (hunk 26 above)
Comment (gregw) [inline testwebapp/src/main/java/com/google/apphosting/tests/usercode/testservlets/DumpServlet.java]: No it didn't.  Ah but I found an option in the eclipse Clean Up profiles.   Trying that out....

**Comment by gregw** [comment]
This is the result of running the eclipse clean up tool, which is for some reason separate to their code formatting tool.

**Comment by aozarov** [comment]
Looks great. Thank you for doing that!

**Pull request merged**